examples/lighting-app/efr32/src/AppTask.cpp

```diff
@@ -47,7 +50,7 @@
 
 #define FACTORY_RESET_TRIGGER_TIMEOUT 3000
 #define FACTORY_RESET_CANCEL_WINDOW_TIMEOUT 3000
-#define APP_TASK_STACK_SIZE (2048)
+#define APP_TASK_STACK_SIZE (1536)
```

Comment:
How do we know that the new, reduced stack sizes are sufficient?

Reply:
With the `MemMonitoring` task we print the minimum available heap ever task by task. After completing the full provisioning flow I took those numbers and reduced them as much as possible leaving between 256-512 bytes of free spaces for future features. Exception made to the OpenThread Task and LWIP which I barely reduced since I didn't test the worst case scenario with them.

Reply:
I'm confused then.  How does tracking max heap allocation for a task tell us about its stack high-water mark?

Reply:
with FreeRTOS api `uxTaskGetStackHighWaterMark` you actually get the task high-water mark. But to actually get a "valid" number you need to identify the worst case scenario and go through at run time. This is how I got these numbers.